### PR TITLE
Mrc-2384 Add note text area to "copy to" dialog in project history

### PR DESCRIPTION
**Pull request by carbowizzy**
This PR adds note field to to copy modal in project history and update promote endpoints accordingly. When a user creates a project from an existing project version, notes text can be added to the project indicating why it was copied. The notes text will be attached to the newly copied project version.

Please describe your changes here

## Type of version change
_Delete as appropriate_

 Minor 

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ProjectsController.kt [hunk 1]
```diff
@@ -80,9 +80,9 @@ class ProjectsController(private val session: Session,
         @RequestParam("name") name: String,
         @RequestParam("note") note: String?): ResponseEntity<String>
     {
-        val newProjectId = projectRepository.saveNewProject(userId(), name, note = note)
+        val newProjectId = projectRepository.saveNewProject(userId(), name)
```
Comment (mwoodbri) [inline src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ProjectsController.kt:83]: So, just to confirm: when creating a new project from an existing project version then the note attached to the version should be transferred to the first version of the new project. And the new project itself won't have its note populated, either from the original project or version?

Comment (carbowizzy) [reply src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ProjectsController.kt:83]: That's right. The project itself won't have its note populated but the version.

**Review by mwoodbri — COMMENTED** [review]
This worked for me and code looks pretty sound: just a couple of comments/clarifications

File: src/app/static/src/app/components/projects/ProjectHistory.vue [hunk 2]
```diff
@@ -273,7 +280,10 @@
             currentLanguage: mapStateProp<RootState, Language>(
                 null,
                 (state: RootState) => state.language
-            )
+            ),
+            projects: mapStateProp<ProjectsState, Project[] | null>(namespace, state => {
```
Comment (EmmaLRussell) [inline src/app/static/src/app/components/projects/ProjectHistory.vue:284]: I don't really understand why this is necessary - the ProjectsMixin was already providing projects.

**Review by EmmaLRussell — CHANGES_REQUESTED** [review]
Works for me! :+1:

I know the integration test is testing the whole process, but could you also add db tests (that the VersionRepository saves the promoted version note, and that the ProjectRepository returns the version notes in getProjects.

**Review by EmmaLRussell — APPROVED** [review]

**Review by mwoodbri — APPROVED** [review]
Good stuff!

**Pull request merged**